createtransfers/createtransfers.py

```diff
@@ -194,17 +199,137 @@ def create_variously_encoded_files():
+        recursion_depth=createtransfersargs.DEFAULT_DEPTH,
+        number_of_directories=createtransfersargs.DEFAULT_NUMBER_DIRS,
+        number_of_files=createtransfersargs.DEFAULT_NUMBER_FILES)
+    # The first directory above is deliberately made one additional directory
+    # down so that it can be zipped as one. First we have to come back up a
+    # level which we do here.
+    one_deep_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                     var_deep_transfer)
+    create_zip_dance(one_deep_zip_path)
+
+    # Create a path where we will create a large zipped transfer folder.
+    zip_path_large = os.path.join(HERE, test_transfers, sample_packages,
+                                  var_large_zip_transfer)
+    # Override the defaults to create a modest 1.1GB zip.
+    create_large_test_transfers(zip_path=zip_path_large,
+                                image_n=35,
+                                video_n=35)
+    # Like the deep transfer, we deliberately drop the large transfers down a
+    # folder level. We come back a level here to zip it all at once.
+    one_large_zip_path = os.path.join(HERE, test_transfers, sample_packages,
+                                      var_large_transfer)
+    create_zip_dance(one_large_zip_path)
+
+
+def create_zip_dance(base_directory):
+    """Perform various manouvers to create zip files from our content and move
+    them into a suitable directory structure.
+    """
+    subdirs = [f for f in os.listdir(base_directory)
+               if os.path.isdir(base_directory)]
+    for dir_ in subdirs:
+        # Create a temporary folder to provide structure inside our zip.
+        sample_package = os.path.join(base_directory, "sample-package")
+        rm_dirs_and_create(sample_package)
+        dir_ = os.path.join(base_directory, dir_)
+        # Move the folder we want to zip underneath the temporary folder.
+        shutil.move(dir_, sample_package)
+        # Re-create the previous folder as we will want to put the final zip
+        # file into here.
+        rm_dirs_and_create(dir_)
+        # Make an archive from our directory.
+        shutil.make_archive(dir_, 'zip', sample_package)
+        # Move the zip file from our temporary folder.
+        move_from = "{}.zip".format(dir_)
+        shutil.move(move_from, dir_)
+        # Finally remove the temporary folder.
+        rm_dirs_and_create(sample_package, True)
+
+
+def create_large_test_transfers(zip_path=None, image_n=113, video_n=669):
     """Create large test transfers by copying existing sampledata files
     multiple times to specific subdirectories of
     TestTransfers/acceptance-tests/performance/
     """
-    image_n = 113
-    video_n = 669
-
     # Create a location for our large transfer.
     performance_path = os.path.join(
         HERE, 'TestTransfers', 'acceptance-tests', 'performance')
+    if zip_path:
+        performance_path = os.path.join(zip_path)
```

Comment:
Just occurs to me that this overriding of `performance_path` may be more general than the context of creating a zip archive. If we may want to override the path used here in other contexts, maybe we should just have a kwarg `performance_path=None` and set it in the standard way if it is not set by the caller. Does that make sense? Just an idea.

Reply:
It might read better than using `zip_path` so have made that change.